File: Lab3/Zadatak1/parrot.c
#include <stdlib.h>

typedef char const* (*PTRFUN)();

struct Parrot {
    PTRFUN* vtable;
    char const* name;
};

char const* name(void* this) {
    return ((struct Parrot*)this)->name;
}

char const* greet() {
    return "Pozdrav! Ja sam papiga";
}

char const* menu() {
    return "Pticja hrana";
}

PTRFUN parrotVTable[3] = {
    (PTRFUN)name,
    (PTRFUN)greet,
    (PTRFUN)menu};

void constructParrot(struct Parrot* this, char const* name) {
    this->name = name;
    this->vtable = parrotVTable;
}

void* create(char const* name) {
    struct Parrot* this = malloc(sizeof(struct Parrot));
    constructParrot(this, name);
    return this;
}

File: Lab3/Zadatak1/Zadatak1.4/src/hr/fer/zemris/ooup/lab2/model/plugins/Parrot.java
package hr.fer.zemris.ooup.lab2.model.plugins;

import hr.fer.zemris.ooup.lab2.model.Animal;

public class Parrot extends Animal {

    private String animalName;

    public Parrot(String name) {
		this.animalName = name;
    }

    @Override
    public String name() {
        return animalName;
    }

    @Override
    public String greet() {
        return "Pozdrav! Ja sam papiga.";
    }

    @Override
    public String menu() {
        return "Pticja hrana.";
    }
}